src/gui/widgets/FadeButton.cpp

```diff
@@ -82,21 +111,51 @@ void FadeButton::customEvent( QEvent * )
 void FadeButton::paintEvent( QPaintEvent * _pe )
 {
 	QColor col = m_normalColor;
+
 	if( ! m_stateTimer.isNull() && m_stateTimer.elapsed() < FadeDuration )
 	{
+		// The first part of the fade, when a note is triggered.
 		const float state = 1 - m_stateTimer.elapsed() / FadeDuration;
-		const int r = (int)( m_normalColor.red() *
+		const int r = (int)( m_holdColor.red() *
 					( 1.0f - state ) +
 			m_activatedColor.red() * state );
-		const int g = (int)( m_normalColor.green() *
+		const int g = (int)( m_holdColor.green() *
 					( 1.0f - state ) +
 			m_activatedColor.green() * state );
-		const int b = (int)( m_normalColor.blue() *
+		const int b = (int)( m_holdColor.blue() *
 					( 1.0f - state ) +
 			m_activatedColor.blue() * state );
 		col.setRgb( r, g, b );
 		QTimer::singleShot( 20, this, SLOT( update() ) );
 	}
+	else if ( ! m_stateTimer.isNull()
+		&& m_stateTimer.elapsed() >= FadeDuration
+		&& activeNotes > 0)
+	{
+		// The fade is done, but at least one note is still held.
+		col = m_holdColor;
+	}
+	else if ( ! m_releaseTimer.isNull() && m_releaseTimer.elapsed() < FadeDuration )
+	{
+		// Last note just ended. Fade to default color.
```

Comment:
This is a duplicate of lines 118:128. Please extract it into a function (something like `fadeToColor(color, timer)`.

Reply:
Done.